### PR TITLE
Add synonyms search

### PR DESCRIPTION
**Pull request by patrick-alaspaa**

### CODE REVIEW

File: pta-intelligent-search-api/src/main/java/fi/maanmittauslaitos/pta/search/api/search/OntologyElasticsearchQueryProviderImpl.java [hunk 1]
```diff
@@ -248,6 +253,28 @@ private QueryBuilder freetextQuery(String field, String word, double weight, dou
...
+    private void addSynonymQueries(Collection<String> searchTerms, BoolQueryBuilder boolQuery) {
+        for (String searchTerm : createSynonymList(searchTerms)) {
+            addFreeTextQuery(searchTerm, boolQuery);
+        }
+    }
+
+    private List<String> createSynonymList(Collection<String> searchTerms) {
+        List<String> synonymList = new ArrayList<>();
+        for(String term : searchTerms) {
+            List<String> termSynonyms = synonymsWords.get(term);
```
Comment (sampov2) [inline pta-intelligent-search-api/src/main/java/fi/maanmittauslaitos/pta/search/api/search/OntologyElasticsearchQueryProviderImpl.java:265]: style issue: getters are preferred over direct field access - helps reader keep track of object-level fields

File: pta-intelligent-search-api/src/main/java/fi/maanmittauslaitos/pta/search/api/search/OntologyElasticsearchQueryProviderImpl.java [hunk 2]
```diff
@@ -248,6 +253,28 @@ private QueryBuilder freetextQuery(String field, String word, double weight, dou
...
+            if(termSynonyms != null && termSynonyms.size() > 0) {
+                for (String synonym : termSynonyms) {
+                	if(synonym != term) {
+						synonymList.add(synonym);
```
Comment (sampov2) [inline pta-intelligent-search-api/src/main/java/fi/maanmittauslaitos/pta/search/api/search/OntologyElasticsearchQueryProviderImpl.java:269]: Indentation issues (here and in other places)

File: pta-intelligent-search-api/src/main/java/fi/maanmittauslaitos/pta/search/api/search/OntologyElasticsearchQueryProviderImpl.java [hunk 3]
```diff
@@ -248,6 +253,28 @@ private QueryBuilder freetextQuery(String field, String word, double weight, dou
...
+					}
+                }
+                //synonymList.remove(term);
```
Comment (sampov2) [inline pta-intelligent-search-api/src/main/java/fi/maanmittauslaitos/pta/search/api/search/OntologyElasticsearchQueryProviderImpl.java:272]: Why is this commented? My intuition is that the original search term that was expanded to synonyms should not appear as a synonym to itself.

File: pta-intelligent-search-api/src/main/java/fi/maanmittauslaitos/pta/search/api/search/OntologyElasticsearchQueryProviderImpl.java (hunk 1 above)
Comment (patrick-alaspaa) [reply pta-intelligent-search-api/src/main/java/fi/maanmittauslaitos/pta/search/api/search/OntologyElasticsearchQueryProviderImpl.java]: Added a getter for synonymWords field

File: pta-intelligent-search-api/src/main/java/fi/maanmittauslaitos/pta/search/api/search/OntologyElasticsearchQueryProviderImpl.java (hunk 2 above)
Comment (patrick-alaspaa) [reply pta-intelligent-search-api/src/main/java/fi/maanmittauslaitos/pta/search/api/search/OntologyElasticsearchQueryProviderImpl.java]: Indentation issues were not visible for me in intellij, but I used another text editor to fix what I could find.

**Pull request merged**